analyzers/src/SonarAnalyzer.CSharp/Rules/SillyBitwiseOperation.cs

```diff
@@ -64,5 +84,101 @@ private void CheckBinary(SonarSyntaxNodeReportingContext context, int constValue
             var binary = (BinaryExpressionSyntax)context.Node;
             CheckBinary(context, binary.Left, binary.OperatorToken, binary.Right, constValueToLookFor);
         }
+
+        private void CheckComparison(SonarSyntaxNodeReportingContext context)
+        {
+            var binary = (BinaryExpressionSyntax)context.Node;
+            if (ShouldRaise(context.SemanticModel, binary.Left, binary.Right, binary.Kind(), false))
+            {
+                context.ReportIssue(Diagnostic.Create(BitwiseRule, binary.GetLocation()));
```

Comment:
Please revert the changes to the Bitwise rule for now. Let's live with duplication until we really see how the solution looks like. We can think about reuse then.